frontend/src/components/LoginForm.tsx

```diff
@@ -1,29 +1,63 @@
-import { Form, Button } from 'react-bootstrap';
-import { Link } from 'react-router-dom';
-import { Container } from 'react-bootstrap';
+/* eslint-disable no-unused-expressions */
+import { useCallback } from 'react';
+import { useHistory, Link } from 'react-router-dom';
+import { Form, Button, Container } from 'react-bootstrap';
+import jwt from 'jwt-decode';
+import loginService from '../services/login';
+import { useUser } from '../context/UserContext';
 
-const LecturerLogin = (props: { isLecturer: boolean }) => {
-  const handleSubmit = (event: any) => {
+interface User {
+  email: string;
+  role: string;
+}
+
+const LoginForm = (props: { isLecturer: boolean }) => {
+  const { isLecturer } = props;
+  const { updateUser } = useUser();
+  const history = useHistory();
+
+  const onUserChange = useCallback((userToUpdate: User) => {
+    updateUser(userToUpdate);
+  }, []);
+
+  const handleSubmit = async (event: any) => {
     event.preventDefault();
-    console.log(event.target.email.value);
-    console.log(event.target.password.value);
 
-    // login logic
+    const credentials = {
+      email: event.target.email.value,
+      password: event.target.password.value,
+    };
+
+    try {
+      const response = await loginService.login(credentials);
+      const decodedUser: { email: string; role: string } = jwt(response.token);
+
+      const userToUpdate: User = {
+        email: decodedUser.email,
+        role: decodedUser.role,
+      };
 
-    event.target.email.value = '';
-    event.target.password.value = '';
+      await onUserChange(userToUpdate);
+
+      window.sessionStorage.setItem('jwt', response.token);
+
+      props.isLecturer ? history.push('/lecturer/dashboard') : history.push('/student/dashboard');
```

Comment:
zamiast propsow ja bym to tez przeniosl do `UserContext`.

Wtedy mozna tez miec jeden link `dashboard`, ktory bedzie sie zachowywal roznie w zaleznosci od `UserContext.isLecturer`